### PR TITLE
Log on Broken Doc Shapes

### PR DESCRIPTION
**Pull request by kstich**
Instead of skipping Code Snippets that have issues, track them and add an option to log them out to a file.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,7 +91,7 @@ api: api-get-apigen
 
 api-models:
 	# Build custom docs
-	php build/docs.php
+	php build/docs.php $(if $(ISSUE-LOGGING-ENABLED),--issue-logging-enabled,)
```
Comment (imshashank) [inline Makefile]: Can you please rename ISSUE-LOGGING-ENABLED to ISSUE_LOGGING_ENABLED

more suited for an env variable format.

File: build/docs/classes/DocsBuilder.php [hunk 2]
```diff
@@ -182,6 +193,24 @@ private function updateQuickLinks(array $services)
...
+                    if (!empty($text)) {
+                        $text .= PHP_EOL;
+                    }
+                    $text .= $serviceName . '-' . $serviceVersion
+                        . ': ' . $message;
+                }
+            }
+        }
+
+        return (bool) file_put_contents("{$this->outputDir}/issues.txt", $text);
```
Comment (imshashank) [inline build/docs/classes/DocsBuilder.php]: would suggest better renaming like error-issues.log

File: build/docs/classes/DocsBuilder.php [hunk 3]
```diff
@@ -711,17 +740,47 @@ private function createHtmlForOperation(Service $service, $name, Operation $oper
...
+                $generatorIssues
+            );
         }
 
         $html->close(); // operation-container
 
         return $html;
     }
 
+    private function logIssues($serviceName, $serviceVersion, $issuesToLog)
```
Comment (imshashank) [inline build/docs/classes/DocsBuilder.php]: Suggestion: Use the standard log issue format example:->
`date LEVEL Issue`

**Comment by imshashank** [comment]
Is there a reason this is reducing code coverage?

**Comment by kstich** [comment]
This doesn't touch any of those files, I think this is an issue with Codecov instead.

**Pull request merged**